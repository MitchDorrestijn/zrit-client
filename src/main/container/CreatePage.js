/**
 * React related imports
 */
import React from 'react';
import {
	Container,
	Row,
	Col
} from 'reactstrap';

/**
 * Self made components imports
 */
import Header from '../components/Header';
import Footer from '../components/Footer';
import CreateZorginstelling from '../views/zorginstelling/CreateZorginstelling';
import CreateClient from '../views/client/CreateClient';
import CreateChauffeur from '../views/chauffeur/CreateChauffeur';

/**
 * This component renders the create page for the elements based on the URL
 */
const CreatePage = (props) => {
	return (
    <Container>
      <Row>
        <Col>
          <Header routes={props.routes} history={props.history} />
						{props.match.url === props.routes.createZorginstelling &&
							<CreateZorginstelling routes={props.routes} history={props.history} />
						}
						{props.match.url === props.routes.createClient &&
							<CreateClient routes={props.routes} history={props.history} />
						}
						{props.match.url === props.routes.createChauffeur &&
							<CreateChauffeur routes={props.routes} history={props.history} />
						}
          <Footer />
        </Col>
      </Row>
    </Container>
	);
}

export default CreatePage;
