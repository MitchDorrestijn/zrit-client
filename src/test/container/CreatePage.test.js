import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import CreatePage from '../../main/container/CreatePage';

const wrapper = shallow(<CreatePage match="" routes="" />);

test('<CreatePage /> should match snapshot', () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('<CreatePage /> should render one <Container> component', () => {
  expect(wrapper.find('Container').length).toBe(1);
});

test('<CreatePage /> should render one <Row> component', () => {
  expect(wrapper.find('Row').length).toBe(1);
});

test('<CreatePage /> should render one <Col> component', () => {
  expect(wrapper.find('Col').length).toBe(1);
});

test('<CreatePage /> should render one <Header> component', () => {
  expect(wrapper.find('Header').length).toBe(1);
});

test('<CreatePage /> should render one <Footer> component', () => {
  expect(wrapper.find('Footer').length).toBe(1);
});
