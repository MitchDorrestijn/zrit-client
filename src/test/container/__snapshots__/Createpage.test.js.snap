// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreatePage /> should match snapshot 1`] = `
<Container
  tag="div"
>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Header
        routes=""
      />
      <CreateZorginstelling
        routes=""
      />
      <Footer />
    </Col>
  </Row>
</Container>
`;
